### ISSUE TITLE
Possible false positive with sort!

### ISSUE THREAD

**Issue by astrozot** [aviatesk/JET.jl#522]
The following code produces an error

````
julia> x = [3,4,5];

julia> @test_call sort!(x)
JET-test failed at REPL[415]:1
  Expression: #= REPL[415]:1 =# JET.@test_call sort!(x)
  ═════ 2 possible errors found ═════
  ┌ @ sort.jl:1367 Base.Sort.:(var"#sort!#28")(Base.Sort.defalg(v), Base.Sort.isless, Base.Sort.identity, Base.Sort.nothing, Base.Sort.Forward, Base.Sort.nothing, #self#, v)
  │┌ @ sort.jl:1374 Base.Sort._sort!(v, Base.Sort.maybe_apply_initial_optimizations(alg), Base.Sort.ord(lt, by, rev, order), NamedTuple{(:scratch,)}(tuple(scratch)))
  ││┌ @ sort.jl:569 Base.Sort._sort!(Core.kwcall(NamedTuple{(:unsafe,)}(tuple(true)), Base.Sort.WithoutMissingVector, v), a.next, o, merge(merge(Base.NamedTuple(), kw), NamedTuple{(:lo, :hi)}(tuple(lo, hi))))
  │││┌ @ sort.jl:660 Base.Sort._sort!(v, a.next, o, kw)
  ││││┌ @ sort.jl:715 Base.Sort._sort!(v, a.big, o, kw)
  │││││┌ @ sort.jl:644 Base.Sort._sort!(v, a.next, o, kw)
  ││││││┌ @ sort.jl:691 Base.Sort._sort!(v, a.yes, o, kw)
  │││││││┌ @ sort.jl:715 Base.Sort._sort!(v, a.big, o, kw)
  ││││││││┌ @ sort.jl:783 Base.Sort._sort!(v, a.next, o, kw)
  │││││││││┌ @ sort.jl:810 Base.Sort._sort!(v, a.next, o, merge(merge(Base.NamedTuple(), kw), NamedTuple{(:mn, :mx)}(tuple(mn, mx))))
  ││││││││││┌ @ sort.jl:836 Base.Sort._sort!(v, a.next, o, kw)
  │││││││││││┌ @ sort.jl:894 Base.Sort._sort!(v, a.radix, o, kw)
  ││││││││││││┌ @ sort.jl:930 u = Base.Sort.uint_map!(v, lo, hi, o)
  │││││││││││││┌ @ sort.jl:1872 u = Base.Sort.reinterpret(Base.Sort.UIntMappable(Base.Sort.eltype(v), order), v)
  ││││││││││││││┌ @ reinterpretarray.jl:75 readable = Base.array_subpadding(T, S)
  │││││││││││││││┌ @ reinterpretarray.jl:732 Base.indexed_iterate(iterate(t), 1)
  ││││││││││││││││┌ @ tuple.jl:91 x = iterate(I)
  │││││││││││││││││ no matching method found `iterate(::Nothing)`: x = iterate(I::Nothing)
  ││││││││││││││││└───────────────
  │││││││││││││││┌ @ reinterpretarray.jl:732 Base.indexed_iterate(iterate(t), 2, _4)
  ││││││││││││││││┌ @ tuple.jl:96 x = iterate(I, state)
  │││││││││││││││││ no matching method found `iterate(::Nothing, ::Int64)`: x = iterate(I::Nothing, state::Int64)
  ││││││││││││││││└───────────────
  
ERROR: There was an error during testing
````

Is this a genuine problem with the implementation of `sort!` or is a JET false positive?

**Comment by astrozot** [comment]
I am seeing other possible false positives: for example `@test_call sum(rand(3,2), dims=2)` also reports an error.

**Comment by aviatesk** [comment]
They can be fixed on the implementation side. We can always try to fix it from the compiler side, but it will require more work.

**Comment by aviatesk** [comment]
> I am seeing other possible false positives: for example `@test_call sum(rand(3,2), dims=2)` also reports an error.

This error is because the syntax does not allow JET to use the constant information of `dims=2`. The error will not be reported if you do
```julia
julia> test_call((Matrix{Float64},)) do a
           sum(a, dims=2)
       end
Test Passed
```

**Comment by aviatesk** [comment]
The `sort!` case will be fixed in JET v0.8.
